wrapper/src/main/java/software/amazon/jdbc/targetdriverdialect/TargetDriverDialectManager.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.targetdriverdialect;
+
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.function.Function;
+import java.util.logging.Logger;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import software.amazon.jdbc.AwsWrapperProperty;
+import software.amazon.jdbc.util.Messages;
+import software.amazon.jdbc.util.StringUtils;
+
+public class TargetDriverDialectManager implements TargetDriverDialectProvider {
+
+  private static final Logger LOGGER = Logger.getLogger(TargetDriverDialectManager.class.getName());
+
+  protected static TargetDriverDialect customDialect;
+
+  public static final AwsWrapperProperty TARGET_DRIVER_DIALECT = new AwsWrapperProperty(
+      "wrapperTargetDriverDialect", "",
+      "A unique identifier for the target driver dialect.");
+
+  /**
+   * Every Dialect implementation SHOULD BE stateless!!!
+   * Dialect objects are shared between different connections.
+   * The order of entries in this map is the order in which dialects are called/verified.
+   */
+  protected static final Map<String, TargetDriverDialect> knownDialectsByCode =
```

Comment:
Nit:  would it be better to have this mapping located in the `TargetDriverDialectCodes` class? That way if a new dialect is introduced, it is one less class that needs to be updated, and it is harder for the developer to forget adding this mapping. 

Also sorry about the lateness of the comment.